server/src/main/java/com/google/coffeehouse/servlets/RetrieveTokenServlet.java

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeTokenRequest;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.coffeehouse.util.AuthenticationConstants;
+import com.google.gson.Gson;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.util.Map;
+import java.security.GeneralSecurityException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to retrieve an ID/auth/refresh token from an auth code and redirect URI input.
+ */
+@WebServlet("/api/retrieve-token")
+public class RetrieveTokenServlet extends HttpServlet {
+  /** The name of the key to be associated with the auth code in the JSON. */
+  public static final String CODE_KEY_NAME = "code";
+  /** The name of the key to be associated with the redirect URI in the JSON. */
+  public static final String REDIRECTURI_KEY_NAME = "redirectUri";
```

Comment:
Nit: Should `REDIRECTURI_KEY_NAME` be `REDIRECT_URI_KEY_NAME`?

Reply:
Yup, absolutely correct. Fix has been pushed.